### PR TITLE
docs: describe importance of enabledBlocking

### PR DESCRIPTION
**Pull request by yjaaidi**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: aio/content/guide/universal.md [hunk 1]
```diff
@@ -320,6 +320,12 @@ For example, if the client requested `https://my-server.com/dashboard` and you a
 
 Now, on every HTTP request made as part of rendering the application on the server, Angular can correctly resolve the request URL to an absolute URL, using the provided `options.url`.
 
+### Optimizing hydration
+
+When the page is loaded in the browser, Angular will bootstrap as usual and replace the content of the DOM. This process is called hydration.
+This hydration process can cause some flicker _(specially for lazy loaded routes)_. In order to avoid the flicker, it is recommended to set the [`initialNavigation`](https://angular.io/api/router/InitialNavigation) option to `enabledBlocking` when using Server-Side Rendering.
+Using this option, Angular will wait for the route to be loaded before bootstrapping and hydrating.
```
Comment (AndrewKushnir) [inline aio/content/guide/universal.md:327]: My understanding is that the NgUniversal schematic patches the `AppRoutingModule` with the `initialNavigation: 'enabledBlocking'`. If so, we can mention it and recommend adding the `initialNavigation` for custom setups.

// cc @alan-agius4

Comment (alan-agius4) [reply aio/content/guide/universal.md:327]: That is correct @AndrewKushnir.

The term hydration here might be misleading as Angular doesn’t re-hydrate the application.

Comment (yjaaidi) [reply aio/content/guide/universal.md:327]: > My understanding is that the NgUniversal schematic patches the `AppRoutingModule` with the `initialNavigation: 'enabledBlocking'`. If so, we can mention it and recommend adding the `initialNavigation` for custom setups.
> 
> 
> 
> // cc @alan-agius4

Oh! Do you know since when this is enabled? I used Universal schematic for the setup on Angular 10 or 11.

Comment (yjaaidi) [reply aio/content/guide/universal.md:327]: > That is correct @AndrewKushnir.
> 
> 
> 
> The term hydration here might be misleading as Angular doesn’t re-hydrate the application. 
> 
> 
> 
> 

Agreed about the misuse of "hydration". Should we talk about "initial rendering"?

Comment (alan-agius4) [reply aio/content/guide/universal.md:327]: > Oh! Do you know since when this is enabled? I used Universal schematic for the setup on Angular 10 or 11.

[9.0.0](https://github.com/angular/universal/releases/tag/v9.0.0)

>  Also, this will happen even more frequently with standalone setups... until the schematic takes them into account.

Universal doesn't support standalone at the moment.

Comment (arturgawlik) [reply aio/content/guide/universal.md:327]: > > That is correct @AndrewKushnir.
> > The term hydration here might be misleading as Angular doesn’t re-hydrate the application.
> 
> Agreed about the misuse of "hydration". Should we talk about "initial rendering"?

actually https://angular.io/guide/roadmap#in-progress calls it "destructive hydration" or "rerendering" :sweat_smile:

Comment (AndrewKushnir) [reply aio/content/guide/universal.md:327]: @yjaaidi I think we can rephrase this as a router configuration and drop "hydration", here is the proposed content:

==============
### Configure Router

To ensure routing is finished before the server serializes the page content, NgUniversal updates the `RouterModule.forRoot` call in your application and adds the `initialNavigation: 'enabledBlocking'` as a config option. This config option also ensures that client-side navigation completes before the application bootstraps in the browser, which allows the page to retain the server-side rendered content until the application is fully rendered.

If you use a custom setup and/or [renderModule](https://angular.io/api/platform-server/renderModule) directly, update the Router configuration manually.

If you use the `RouterModule.forRoot`, include the `initialNavigation` option.

```
RouterModule.forRoot(routes, {initialNavigation: 'enabledBlocking'});
```

If you use the `provideRouter` API, add the `withEnabledBlockingInitialNavigation` function call.

```
provideRouter(routes, withEnabledBlockingInitialNavigation());
```

==============

What do you think?

Comment (yjaaidi) [reply aio/content/guide/universal.md:327]: Hey @AndrewKushnir ! Sorry for getting back to you so late.
I was off for a while and now I'm catching up 😅

I like your suggestion. Let's go ahead.
Would you want me to update my PR with it?

Comment (AndrewKushnir) [reply aio/content/guide/universal.md:327]: Hey @yjaaidi, no problem!

With the upcoming non-destructive hydration and some changes that we've made in the platform-server, the `enabledBlockingInitialNavigation` option would no longer be required. I think at this point we can just leave the docs as is (i.e. close this PR) and we'll eventually deprecate the `enabledBlockingInitialNavigation` once the non-destructive hydration exits the developer preview.

Comment (yjaaidi) [reply aio/content/guide/universal.md:327]: Perfect! Let's do that.
Sorry for the noise then 😅
